src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -271,8 +271,14 @@ private void ReadSettings(PackageRestoreInputs packageRestoreInputs)
                     GetInstalledPackageReferences(packageReferenceFile, allowDuplicatePackageIds: true));
             }
 
+            // If PackageSaveMode.None is provided, this means that the user did not specify a
+            // PackageSaveMode. In the case of V2 restore, this should default to PackageSaveMode.Defaultv2.
+            var packageSaveMode = EffectivePackageSaveMode == Packaging.PackageSaveMode.None ?
```

Comment:
No, the value returned by `CalculateEffectivePackageSaveMode` is used in both V2 and V3 restores. `None` has the dual meaning of `Defaultv2` for the V2 restore step and `Defaultv3` in the V3 restore step.